src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -75,15 +75,53 @@ type commitLogSource struct {
 	snapshotFilesFn snapshotFilesFn
 	newReaderFn     newReaderFn
 
-	metrics commitLogSourceDataAndIndexMetrics
+	metrics commitLogSourceMetrics
+}
+
+type bootstrapNamespace struct {
+	namespaceID             []byte
+	bootstrapping           bool
+	dataAndIndexShardRanges result.ShardTimeRanges
+	namespace               namespace.Metadata
+	namespaceContext        namespace.Context
+	dataBlockSize           time.Duration
+	accumulator             bootstrap.NamespaceDataAccumulator
+}
+
+type seriesMap map[seriesMapKey]*seriesMapEntry
+
+type seriesMapKey struct {
+	fileReadID  uint64
+	uniqueIndex uint64
+}
+
+type seriesMapEntry struct {
+	namespace *bootstrapNamespace
+	series    bootstrap.CheckoutSeriesResult
 }
 
-type encoder struct {
-	lastWriteAt time.Time
-	enc         encoding.Encoder
+// accumulateArg contains all the information a worker go-routine needs to encode
```

Comment:
nit: this comment is a bit messed up, maybe this?
```golang
// accumulateArg contains all the information a worker go-routine needs to
// accumulate a write for encoding.
```

Reply:
Sure thing.